tests/i18n/test_extraction.py

```diff
@@ -127,6 +127,33 @@ def assertNotRecentlyModified(self, path):
         self.assertGreater(delta, 10, "%s wasn't recently modified" % path)
 
 
+class NormalizeLocaleTest(ExtractorTests):
+
+    def test_makemessages_with_normalized_locale(self):
+        """
+        Test the makemessages works with normalized locales in the input.
+        """
+
+        for locale in ('zh-cn', 'zh_cn', 'zh_CN', 'ZH-CN',):
+            out = StringIO()
+
+            with self.subTest():
+                # Test the Normalization of different strings used for locales
+                locale_normalized = normalize_locale(locale, out)
+                self.assertEqual('zh_CN', locale_normalized)
+
+            with self.subTest():
+                if locale in ('zh-CN', 'zh_CN', 'ZH-CN',):
+                    # Test the warning given on normalizing the locale
+                    self.assertIn('locale normalized to zh-cn', out.getvalue())
+                elif locale in ('zh-cn',):
+                    # Locale normalization is not needed
+                    self.assertIn('', out.getvalue())
```

Comment:
This assertion will always pass... The empty string is always in any other string...
You should reset the `StringIO` buffer in each loop, check for equality, and the whole block can be simplified, perhaps:
```python
self.assertEqual('Normalized %s to zh_CN.\n' % locale if locale != 'zh_CN' else '', out.getvalue())
```